Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.cpp

```diff
@@ -48,6 +48,7 @@ using namespace WebCore;
 NicosiaImageBufferPipeSource::NicosiaImageBufferPipeSource()
 {
     m_nicosiaLayer = Nicosia::ContentLayer::create(*this);
+    m_nicosiaLayer->proxy().setContentType(TextureMapperPlatformLayerProxy::ContentType::OffscreenCanvas);
```

Comment:
Ah, it's OffscreenCanvas but set after construction, right?. It's unfortunate we can't set it on construction. Maybe it would be less confusing to make `TextureMapperPlatformLayerProxyGL` have a default value to undefined and not give any argument in `Nicosia::ContentLayer::create()`.

Reply:
Or can we use the `Nicosia::ContentLayer::create()` that receives the proxy instead? And we don't need the setter, making it a construct only property.

Reply:
> Or can we use the `Nicosia::ContentLayer::create()` that receives the proxy instead? And we don't need the setter, making it a construct only property.

I think that's a good idea, but I think then we should remove the `Nicosia::ContentLayer::create()` version that doesn't receive a proxy to ensure that the proxies need to be specifically created with a concrete contentType.

Reply:
Let's do it if it's unused.